libraries/AP_Eigen/AP_Algebra.h

```diff
@@ -0,0 +1,165 @@
+#ifndef AP_EXTH_H
+#define AP_EXTH_H
+
+#include <limits>
+#include <type_traits>
+#include <float.h>
+
+#undef min
+#undef max
+
+/*
+ * @brief: Safe way to convert floating point numbers to integers
+ * @authors: Daniel Frenzel <dgdanielf@gmail.com>
```

Comment:
I like to give credit where it's due, but adding it to each function is pointless. What happens when some one else modifies the function?  You could add your copyright to the beginning of the file, because you wrote it all (or a great part of it when people start changing it); but let's keep the code clean: git is awesome to return the authorship of each and every line in the codebase, so no need to add it here.